zproject/backends.py

```diff
@@ -899,6 +900,77 @@ def external_auth_method(cls: Type[ExternalAuthMethod]) -> Type[ExternalAuthMeth
     EXTERNAL_AUTH_METHODS.append(cls)
     return cls
 
+ExternalAuthDataDict = TypedDict('ExternalAuthDataDict', {
+    'subdomain': str,
+    'full_name': str,
+    'email': str,
+    'is_signup': bool,
+    'redirect_to': str,
+    'mobile_flow_otp': Optional[str],
+    'desktop_flow_otp': Optional[str],
+    'multiuse_object_key': str,
```

Comment:
This should be optional, right?

Reply:
Could be a good idea in a follow up commit perhaps? I tried to stay consistent with how login_or_register handled the "no value" representation - and it was inconsistent in that:
```
def login_or_register_remote_user(request: HttpRequest, email: str,
                                  user_profile: Optional[UserProfile], full_name: str='',
                                  mobile_flow_otp: Optional[str]=None,
                                  desktop_flow_otp: Optional[str]=None,
                                  is_signup: bool=False, redirect_to: str='',
                                  multiuse_object_key: str='',
                                  full_name_validated: bool=False) -> HttpResponse:
```

otps were ``Optional`` with ``None`` meaning no value, but ``multiuse_object_key`` used ``''`` as "no value".